packages/opal-server/opal_server/authentication/authenticator.py

```diff
@@ -0,0 +1,69 @@
+from typing import Optional
+
+from fastapi import Header
+from fastapi.exceptions import HTTPException
+from opal_common.authentication.authenticator import Authenticator
+from opal_common.authentication.deps import JWTAuthenticator
+from opal_common.authentication.oauth2 import (
+    CachedOAuth2Authenticator,
+    OAuth2ClientCredentialsAuthenticator,
+)
+from opal_common.authentication.signer import JWTSigner
+from opal_common.authentication.types import JWTClaims
+from opal_common.authentication.verifier import JWTVerifier, Unauthorized
+from opal_common.config import opal_common_config
+from opal_common.logger import logger
+from opal_server.config import opal_server_config
+
+
+class _ServerAuthenticator(Authenticator):
+    def __init__(self):
+        if opal_common_config.AUTH_TYPE == "oauth2":
```

Comment:
Same comments as in `_ClientAuthenticator`

Reply:
Refactored.